internal/tool/kubectl.go

```diff
@@ -94,49 +97,80 @@ func NewKubectl(kubeConfig clientcmd.ClientConfig) (*Kubectl, error) {
 	return kubectl, nil
 }
 
-func (k Kubectl) WaitForDeployments(context context.Context, namespace string, selector string) error {
+func (k Kubectl) WaitForWorkloadResources(context context.Context, namespace string, selector string) error {
 	deadline, _ := context.Deadline()
-	unavailableDeployments := []deploymentNotReady{{Name: "none", Unavailable: 1}}
-	getDeploymentsError := ""
-
-	utils.LogInfo(fmt.Sprintf("Start wait for deployments. --timeout time left: %s ", time.Until(deadline).String()))
-
-	for deadline.After(time.Now()) && len(unavailableDeployments) > 0 {
-		unavailableDeployments = []deploymentNotReady{}
-		deployments, err := listDeployments(k, context, namespace, selector)
-		if err != nil {
-			unavailableDeployments = []deploymentNotReady{{Name: "none", Unavailable: 1}}
-			getDeploymentsError = fmt.Sprintf("error getting deployments from namespace %s : %v", namespace, err)
-			utils.LogWarning(getDeploymentsError)
+	unavailableWorkloadResources := []workloadNotReady{{Name: "none", Unavailable: 1}}
+	getWorkloadResourceError := ""
+
+	// Loop until timeout reached or all requested pods are available
+	utils.LogInfo(fmt.Sprintf("Start wait for workloads resources. --timeout time left: %s ", time.Until(deadline).String()))
+	for deadline.After(time.Now()) && len(unavailableWorkloadResources) > 0 {
+		unavailableWorkloadResources = []workloadNotReady{}
+
+		deployments, errDeployments := listDeployments(k, context, namespace, selector)
+		daemonSets, errDaemonSets := listDaemonSets(k, context, namespace, selector)
+		statefulSets, errStatefulSets := listStatefulSets(k, context, namespace, selector)
```

Comment:
So I'm understanding that we check for these three resources regardless of whether the deployed chart uses them? The idea being that charts that don't use statefulsets/daemonsets would just auto-pass the "readiness" check because none of those workloads exist. Is that a correct understanding?

That's fine, I just want to make sure. Fundamentally, this check seems to be flawed because charts can deploy _all sorts_ of resources. But at the end of the day, this is a better check than nothing.

Reply:
yes, charts that don't try do deploy certain resources will just auto-pass since we're not expecting them to be there.

As far as checking for other resources I could add in jobs at least because it seems like the only one we're missing from the list of default workload resources (https://kubernetes.io/docs/concepts/workloads/) but didn't know if i should since it wasn't requested and didn't want to go overboard. Happy to make another PR for that though if it's needed